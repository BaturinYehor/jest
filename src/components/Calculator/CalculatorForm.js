import React from "react";

class CalculatorForm extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            redirectToMain: false,
            response: [],
            error: [],
            message: ""
        }
    };

    calculate = (e) => {
        e.preventDefault();

        let x = e.target['x'].value;
        let y = e.target['y'].value;

        if (!isNaN(x) && !isNaN(y)) {
            if (isNaN(x) && isNaN(y)) {
                throw new Error('Not x neither y was set');
            } else {
                throw new Error('Not x neither y was set');
            }
        } else {
            this.setState({message: x + y})
        }
    };

    render() {
        return (
            <div>
                <form onSubmit={this.calculate}>
                    <div className="get-name-form">
                        <div className="row">
                            <div className="board">
                                <div className="text">
                                    fill in the fields, press the button and get the result - a number, if the fields
                                    are
                                    filled with numbers, a string-concatenation of characters, if the fields are filled
                                    with
                                    characters
                                </div>
                            </div>
                        </div>
                        <div className="row">
                            <div className="col-sm-6">
                                <input type="text" name="x" placeholder="x"/>
                            </div>
                            <div className="col-sm-6">
                                <div className="board">
                                    <div className="text">
                                        {this.state.message}
                                    </div>
                                </div>
                            </div>
                        </div>
                        <div className="row">
                            <div className="col-sm-6">
                                <input type="text" name="y" placeholder="y"/>
                            </div>
                            <div className="col-sm-6">
                                <div className="get-name-button">
                                    <button>Calculate</button>
                                </div>
                            </div>
                            {this.state.error}
                        </div>
                        <div className="row h-50">
                            <div className="col-sm-9">
                            </div>
                            <div className="col-sm-1">
                                <div className="img"/>
                            </div>
                        </div>
                    </div>
                </form>
            </div>
        )
    };
}

export default CalculatorForm