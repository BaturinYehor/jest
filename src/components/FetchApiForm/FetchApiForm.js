import React from "react";

const API = 'https://961826c1-030a-4d64-9117-ed8352256d12.mock.pstmn.io/info';

class Main extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            response: [],
            error: []
        }
    };

    getInformation = (e) => {
        e.preventDefault();
        this.fetchApi(API);
    };

    fetchApi(api) {
        fetch(api)
            .then(response => {
                if(response.ok) {
                    response.json()
                        .then((data) => {
                        if (data.info) {
                            this.setState({response: data.info});//information fetched from api
                        } else {
                            this.setState({error: data.error.message})//can't find api to fetch
                        }
                    })
                } else {
                    throw new Error('Error while fetching api');
                }
            })
            .catch((error) => {
                this.setState({error: error})//error while fetching information from api
            });
    }

    render() {
        return (
            <div>
                <button onClick={this.getInformation}>
                    What is jest?
                </button>
                <div className="get-api-form">
                    <div className="text">
                        {this.state.response}
                        <div className="error">
                            {this.state.error}
                        </div>
                    </div>
                </div>
            </div>
        )
    };
}

export default Main