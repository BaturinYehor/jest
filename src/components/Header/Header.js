import React from "react"

class Header extends React.Component {

    buildHelloString() {
        return "Hello " + this.props.name;
    }

    render() {
        return (
            <div className="text">
                {this.buildHelloString()}
            </div>
        );
    }
}

export default Header