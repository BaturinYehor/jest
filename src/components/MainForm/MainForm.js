import React from "react";
import FetchApiForm from "../FetchApiForm/FetchApiForm"
import Header from "../Header/Header"
import CalculatorForm from "../Calculator/CalculatorForm";
import {Redirect} from "react-router";

class MainForm extends React.Component {

    render() {
        if (!this.props.name) {
            return <Redirect to='/greeting'/>
        }
        return (
            <div className="wrapper">
                <div className="main">
                    <div className="container">
                        <div className="row">
                            <div className="col-sm-12">
                                <Header name={this.props.name}/>
                            </div>
                        </div>
                        <div className="row">
                            <div className="col-sm-6">
                                <FetchApiForm/>
                            </div>
                            <div className="col-sm-1">
                                <div className="headerdivider"></div>
                            </div>
                            <div className="col-sm-5">
                                <CalculatorForm/>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        )
    }
}

export default MainForm