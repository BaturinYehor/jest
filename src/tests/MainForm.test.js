import renderer from "react-test-renderer";
import MainForm from "../components/MainForm/MainForm";
import React from "react";
import {mount} from 'enzyme';
import Header from "../components/Header/Header";
import {MemoryRouter} from "react-router-dom/cjs/react-router-dom.min";
import FetchApiForm from "../components/FetchApiForm/FetchApiForm";
import CalculatorForm from "../components/Calculator/CalculatorForm";

describe('GreetingForm test suite', () => {

    it('MainForm should match snapshot', () => {
        const component = renderer.create(<MainForm name={"Nick"}/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('MainForm form should have exactly one Header component', () => {
        const wrapper = mount(<MainForm name={"Nick"}/>);
        expect(wrapper.find(Header).length).toEqual(1);
    });

    it('Should redirect to greeting form', () => {
        const wrapper = mount(
            <MemoryRouter>
                <MainForm/>
            </MemoryRouter>
        );
        const app = wrapper.find(MainForm).instance();
        expect(app.render().props.to).toBe("/greeting")
    });

    it('Should render components', () => {
        const name = "Nick";
        const wrapper = mount(
            <MemoryRouter>
                <MainForm name={name}/>
            </MemoryRouter>
        );
        expect(wrapper.find(Header).length).toEqual(1);
        expect(wrapper.find(FetchApiForm).length).toEqual(1);
        expect(wrapper.find(CalculatorForm).length).toEqual(1);
    });

    it('Should pass props to Header component', () => {
        const name = "Nick";
        const wrapper = mount(
            <MemoryRouter>
                <MainForm name={name}/>
            </MemoryRouter>
        );
        expect(wrapper.find("Header").props().name).toEqual(name);
    });
});