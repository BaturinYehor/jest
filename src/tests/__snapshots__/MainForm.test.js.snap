// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GreetingForm Should match snapshot 1`] = `
<div
  className="wrapper"
>
  <div
    className="main"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-sm-12"
        >
          <div
            className="text"
          >
            Hello Nick
          </div>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-sm-6"
        >
          <div>
            <button
              onClick={[Function]}
            >
              What is jest?
            </button>
            <div
              className="get-api-form"
            >
              <div
                className="text"
              >
                <div
                  className="error"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-sm-1"
        >
          <div
            className="headerdivider"
          />
        </div>
        <div
          className="col-sm-5"
        >
          <div>
            <form
              onSubmit={[Function]}
            >
              <div
                className="get-name-form"
              >
                <div
                  className="row"
                >
                  <div
                    className="board"
                  >
                    <div
                      className="text"
                    >
                      fill in the fields, press the button and get the result - a number, if the fields are filled with numbers, a string-concatenation of characters, if the fields are filled with characters
                    </div>
                  </div>
                </div>
                <div
                  className="row"
                >
                  <div
                    className="col-sm-6"
                  >
                    <input
                      name="x"
                      placeholder="x"
                      type="text"
                    />
                  </div>
                  <div
                    className="col-sm-6"
                  >
                    <div
                      className="board"
                    >
                      <div
                        className="text"
                      >
                        
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  className="row"
                >
                  <div
                    className="col-sm-6"
                  >
                    <input
                      name="y"
                      placeholder="y"
                      type="text"
                    />
                  </div>
                  <div
                    className="col-sm-6"
                  >
                    <div
                      className="get-name-button"
                    >
                      <button>
                        Calculate
                      </button>
                    </div>
                  </div>
                  <div
                    className="error"
                  >
                    
                  </div>
                </div>
                <div
                  className="row h-50"
                >
                  <div
                    className="col-sm-9"
                  />
                  <div
                    className="col-sm-1"
                  >
                    <div
                      className="img"
                    />
                  </div>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
